core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -2059,6 +2059,14 @@ object SnappySession extends Logging {
     if (executedPlan.find(_.isInstanceOf[InMemoryTableScanExec]).isDefined) {
       throw new EntryExistsException("uncached plan", cdf)
     }
+
+    // Try plan caching for only snappy tables
+    if (executedPlan.find(t => t match {
+      case dsc: DataSourceScanExec => !dsc.relation.isInstanceOf[DependentRelation]
+      case _ => false
+    }).isDefined) {
+      throw new EntryExistsException("uncached plan", cdf)
+    }
```

Comment:
Can you combine with the search above instead of a separate searches?

```scala
executedPlan.find {
  case _: InMemoryTableScanExec => true
  case ...
}
```

Reply:
I initially thought so but decided against it for better code clarity. Anyways if plan caching works then this be just done once. For non plan caching cases this is a very small cost. Can do it still if you feel so.

Reply:
Ok, will do. Just saw the code again. Code clarity won't be affected.

Reply:
These small costs have added up to reduce the best performance of point queries from ~2ms at the time of 0.7-perf merge to ~3ms now, so best to avoid this. I don't see any loss of code clarity with merging the two. Add comments in individual matches statements if you think so.